src/build/env.py

```diff
@@ -23,10 +23,15 @@
 else:
     from importlib import metadata
 
+
 try:
     import virtualenv
-except ImportError:  # pragma: no cover
-    virtualenv = None  # pragma: no cover
+except ImportError:
+    virtualenv = None
+else:
+    virtualenv_version = packaging.version.Version(metadata.version('virtualenv'))
+    if virtualenv_version < packaging.version.Version('20.0.35'):
+        virtualenv = None  # pragma: no cover
```

Comment:
I am not too happy with checking the virtualenv version directly. Could we please use `build.check_dependency` to validate `build[virtualenv]`?

Reply:
I would also prefer that this wasn't done on import.  Can we put it inside a function which is then cached when the isolated env builder is first called?

Reply:
If we bump the versions, say `build==0.3.0` suddenly requires `virtualenv>=21`, would `build[virtualenv]` correctly resolve to the current version of build's requirements? And also, does this always pull from the local version or does it need a connection to the package index (might also answer the question above).

Reply:
`check_dependency` is a thin wrapper around `importlib.metadata` and `packaging`:

```
In [1]: from build import check_dependency

In [2]: list(check_dependency('build[virtualenv]'))
Out[2]: [('build[virtualenv]', 'virtualenv>=20.0.35; extra == "virtualenv"')]

In [3]: !python -m pip install virtualenv
Collecting virtualenv
  Using cached virtualenv-20.4.2-py2.py3-none-any.whl (7.2 MB)
Requirement already satisfied: six<2,>=1.9.0 in ./venv/lib/python3.9/site-packages (from virtualenv) (1.15.0)
Requirement already satisfied: appdirs<2,>=1.4.3 in ./venv/lib/python3.9/site-packages (from virtualenv) (1.4.4)
Requirement already satisfied: filelock<4,>=3.0.0 in ./venv/lib/python3.9/site-packages (from virtualenv) (3.0.12)
Requirement already satisfied: distlib<1,>=0.3.1 in ./venv/lib/python3.9/site-packages (from virtualenv) (0.3.1)
Installing collected packages: virtualenv
Successfully installed virtualenv-20.4.2

In [4]: list(check_dependency('build[virtualenv]'))
Out[4]: []

In [5]: !python -m pip install -U 'virtualenv<20.0.35'
Collecting virtualenv<20.0.35
  Downloading virtualenv-20.0.34-py2.py3-none-any.whl (4.9 MB)
     |████████████████████████████████| 4.9 MB 2.8 MB/s
Requirement already satisfied: filelock<4,>=3.0.0 in ./venv/lib/python3.9/site-packages (from virtualenv<20.0.35) (3.0.12)
Requirement already satisfied: six<2,>=1.9.0 in ./venv/lib/python3.9/site-packages (from virtualenv<20.0.35) (1.15.0)
Requirement already satisfied: distlib<1,>=0.3.1 in ./venv/lib/python3.9/site-packages (from virtualenv<20.0.35) (0.3.1)
Requirement already satisfied: appdirs<2,>=1.4.3 in ./venv/lib/python3.9/site-packages (from virtualenv<20.0.35) (1.4.4)
Installing collected packages: virtualenv
  Attempting uninstall: virtualenv
    Found existing installation: virtualenv 20.4.2
    Uninstalling virtualenv-20.4.2:
      Successfully uninstalled virtualenv-20.4.2
Successfully installed virtualenv-20.0.34

In [6]: list(check_dependency('build[virtualenv]'))
Out[6]: [('build[virtualenv]', 'virtualenv>=20.0.35; extra == "virtualenv"')]
```

Reply:
`importlib.metadata` only reads locally installed distributions (specifically, it only checks stuffs in `sys.path`). The overhead should be pretty low, but a lazy implementation is still a good idea IMO.